modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/type/AutoDownloadLimitSetting.kt

```diff
@@ -50,10 +50,12 @@ enum class AutoDownloadLimitSetting(
         fun fromPreferenceString(stringValue: String): AutoDownloadLimitSetting {
             return try {
                 val intValue = stringValue.toInt()
-                entries.firstOrNull { it.preferenceInt == intValue } ?: TWO_LATEST_EPISODE
+                entries.firstOrNull { it.id == intValue } ?: TWO_LATEST_EPISODE
             } catch (e: Exception) {
                 TWO_LATEST_EPISODE
             }
         }
+
+        fun fromInt(id: Int) = (AutoDownloadLimitSetting.entries.firstOrNull { it.id == id } ?: TWO_LATEST_EPISODE)
```

Comment:
Similarly to `fromPreferenceString` I think I'd be better to use any default value in case of missing entries.